pkg/util/kube/utils.go

```diff
@@ -135,3 +137,72 @@ func GetVolumeDirectory(pod *corev1api.Pod, volumeName string, pvcLister corev1l
 
 	return pvc.Spec.VolumeName, nil
 }
+
+// IsCRDReady checks a CRD to see if it's ready, with both the Established and NamesAccepted conditions.
+func IsCRDReady(crd *apiextv1beta1.CustomResourceDefinition) bool {
+	var isEstablished, namesAccepted bool
+	for _, cond := range crd.Status.Conditions {
+		if cond.Type == apiextv1beta1.Established && cond.Status == apiextv1beta1.ConditionTrue {
+			isEstablished = true
+		}
+		if cond.Type == apiextv1beta1.NamesAccepted && cond.Status == apiextv1beta1.ConditionTrue {
+			namesAccepted = true
+		}
+	}
+
+	return (isEstablished && namesAccepted)
+}
+
+// IsUnstructuredCRDReady checks an unstructured CRD to see if it's ready, with both the Established and NamesAccepted conditions.
+// TODO: Delete this function and use IsCRDReady when the upstream runtime.FromUnstructured function properly handles int64 field conversions.
```

Comment:
would it be possible to add a unit test of the `runtime.FromUnstructured` func that currently passes, but would start failing when the issue is fixed? This probably isn't essential, but would give us an active signal for when we can remove this code.

Reply:
Probably, though I'm not sure that it would be directly on this function.

Let me play around with adding one.

Reply:
Added.